### ISSUE TITLE
PanicException: called `Option::unwrap()` on a `None` value on lazy data frame collect

### ISSUE THREAD

**Issue by nesb1** [pola-rs/polars#15883]
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
import polars as pl

df = pl.DataFrame(
        [
            {
                "response": '{"items": [{"data": {"field1":"value","field2": "value"}}]}',
                "vehicle_identifier": "123",
            }
        ]
    )

    structure = pl.List(pl.Struct({"data": pl.Struct({"field1": pl.String, "field2": pl.String})}))
    df.lazy().with_columns(
        identifier=pl.col("vehicle_identifier"),
        items=pl.col("response").str.json_path_match("$.items").str.json_decode(structure),
    ).explode("items").rename({"items": "item"}).with_columns(
        field1=pl.col("item").struct.field("data").struct.field("field1"),
        field2=pl.col("item").struct.field("data").struct.field("field2"),
    ).select(
        "field1", "field2"
    ).collect()

```

### Log output

```shell
thread 'thread 'polars-3polars-2' panicked at ' panicked at crates/polars-lazy/src/physical_plan/expressions/column.rscrates/polars-lazy/src/physical_plan/expressions/column.rs::127127::1414:
:
called `Option::unwrap()` on a `None` valuecalled `Option::unwrap()` on a `None` value

note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/home/dev/sfs-python-sdk/polars_bug.py", line 21, in <module>
    ).collect()
      ^^^^^^^^^
  File "/home/dev/sfs-python-sdk/.hatch/env/virtual/dev.py3.12/lib/python3.12/site-packages/polars/lazyframe/frame.py", line 1700, in collect
    return wrap_df(ldf.collect())
                   ^^^^^^^^^^^^^
pyo3_runtime.PanicException: called `Option::unwrap()` on a `None` value
```

### Issue description

In a new version 0.20.22 this code falls with PanicException. But if I add in last collect call `identifier` or `vehicle_identifier` thic code works without error. Like this:

```python
import polars as pl


df = pl.DataFrame(
    [
        {
            "response": '{"items": [{"data": {"field1":"value","field2": "value"}}]}',
            "vehicle_identifier": "123",
        }
    ]
)

structure = pl.List(pl.Struct({"data": pl.Struct({"field1": pl.String, "field2": pl.String})}))
df.lazy().with_columns(
    identifier=pl.col("vehicle_identifier"),
    items=pl.col("response").str.json_path_match("$.items").str.json_decode(structure),
).explode("items").rename({"items": "item"}).with_columns(
    field1=pl.col("item").struct.field("data").struct.field("field1"),
    field2=pl.col("item").struct.field("data").struct.field("field2"),
).select(
    "field1", "field2", "identifier"
).collect()

It also can be releated with https://github.com/pola-rs/polars/issues/15884

```

### Expected behavior

In an older versions this code works without exeptions and returns `DataFrame`

### Installed versions

<details>

```
--------Version info---------
Polars:               0.20.22
Index type:           UInt32
Platform:             Linux-5.15.0-97-generic-x86_64-with-glibc2.35
Python:               3.12.2 (main, Feb 25 2024, 16:35:05) [GCC 11.4.0]

----Optional dependencies----
adbc_driver_manager:  <not installed>
cloudpickle:          <not installed>
connectorx:           <not installed>
deltalake:            <not installed>
fastexcel:            <not installed>
fsspec:               2024.3.1
gevent:               <not installed>
hvplot:               <not installed>
matplotlib:           <not installed>
nest_asyncio:         <not installed>
numpy:                1.26.4
openpyxl:             <not installed>
pandas:               <not installed>
pyarrow:              16.0.0
pydantic:             1.10.15
pyiceberg:            <not installed>
pyxlsb:               <not installed>
sqlalchemy:           2.0.29
xlsx2csv:             <not installed>
xlsxwriter:           <not installed>
None
```

</details>

**Comment by cmdlineluser** [comment]
Similar to the other issue it seems to be optimizer related.

```python
.collect(comm_subexpr_elim=False)
```

```python
shape: (1, 2)
┌────────┬────────┐
│ field1 ┆ field2 │
│ ---    ┆ ---    │
│ str    ┆ str    │
╞════════╪════════╡
│ value  ┆ value  │
└────────┴────────┘
```

**Comment by mkleinbort-ic** [comment]
I also have this issue.

**Comment by morne** [comment]
+1

**Comment by ritchie46** [comment]
fixed by #15902

**Comment by eromoe** [comment]
I had updated to latest 0.20.25 ,  fix one of my cases .
But today I met another : 
```
thread 'polars-6' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'polars-6' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-6' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-6' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-6' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-6' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-6' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-6' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-6' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-6' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-6' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-6' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-6' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-6' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-6' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-6' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-6' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-6' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-6' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` valuethread '
polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-6' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-6' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-6' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-6' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-6' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-6' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-6' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-6' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
thread 'polars-4' panicked at C:\a\polars\polars\crates\polars-parquet\src\arrow\read\deserialize\mod.rs:144:31:
called `Option::unwrap()` on a `None` value
Traceback (most recent call last):
  File "E:\Workspace\github_me\proj\train_test_model_scroll.py", line 411, in <module>
    train_df = read_partitioned_df(dates_partition[-TRAIN_LOAD_STEP-1:-1], pre_train_partitions, train_df)
  File "E:\Workspace\github_me\proj\train_test_model_scroll.py", line 297, in read_partitioned_df
    train_df = read_feature_df(TRAIN_TIMES, FEATURE_SETS, train_partitions, window=6, sel_features=SEL_FEATURES)
  File "E:\Workspace\github_me\proj\notebook_exports.py", line 1707, in read_feature_df
    train_df = load_multi_partition_features_v3(dts, train_times, feature_sets, verbose=verbose, **kwargs)
  File "E:\Workspace\github_me\proj\notebook_exports.py", line 1571, in load_multi_partition_features_v3
    mfeatures_df = read_partition(path, filters=dates_to_partition_filters(partition_dates))
  File "e:\workspace\github_me\data_dumper\data_dumper\filesystem.py", line 344, in read_partition
    df = df.collect().to_pandas()
  File "C:\envs\quant\lib\site-packages\polars\lazyframe\frame.py", line 1816, in collect
    return wrap_df(ldf.collect(callback))
pyo3_runtime.PanicException: called `Option::unwrap()` on a `None` value
```
